// n8n-nodes-base/dist/nodes/SyncroMSP/v1/actions/versionDescription.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var versionDescription_exports = {};
__export(versionDescription_exports, {
  versionDescription: () => versionDescription
});
module.exports = __toCommonJS(versionDescription_exports);
var import_n8n_workflow = require("n8n-workflow");
var contact = __toESM(require("./contact"));
var customer = __toESM(require("./customer"));
var rmm = __toESM(require("./rmm"));
var ticket = __toESM(require("./ticket"));
const versionDescription = {
  displayName: "SyncroMSP",
  name: "syncroMsp",
  // eslint-disable-next-line n8n-nodes-base/node-class-description-icon-not-svg
  icon: "file:syncromsp.png",
  group: ["output"],
  version: 1,
  subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
  description: "Gets data from SyncroMSP",
  defaults: {
    name: "SyncroMSP"
  },
  inputs: [import_n8n_workflow.NodeConnectionTypes.Main],
  outputs: [import_n8n_workflow.NodeConnectionTypes.Main],
  credentials: [
    {
      name: "syncroMspApi",
      required: true,
      testedBy: "syncroMspApiCredentialTest"
    }
  ],
  properties: [
    {
      displayName: "Resource",
      name: "resource",
      type: "options",
      noDataExpression: true,
      options: [
        {
          name: "Contact",
          value: "contact"
        },
        {
          name: "Customer",
          value: "customer"
        },
        {
          name: "RMM",
          value: "rmm"
        },
        {
          name: "Ticket",
          value: "ticket"
        }
      ],
      default: "contact"
    },
    ...customer.descriptions,
    ...ticket.descriptions,
    ...contact.descriptions,
    ...rmm.descriptions
  ]
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  versionDescription
});
