// n8n-nodes-base/dist/nodes/Postgres/v1/PostgresV1.node.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var PostgresV1_node_exports = {};
__export(PostgresV1_node_exports, {
  PostgresV1: () => PostgresV1
});
module.exports = __toCommonJS(PostgresV1_node_exports);
var import_n8n_workflow = require("n8n-workflow");
var import_descriptions = require("../../../utils/descriptions");
var import_genericFunctions = require("./genericFunctions");
var import_transport = require("../transport");
const versionDescription = {
  displayName: "Postgres",
  name: "postgres",
  icon: "file:postgres.svg",
  group: ["input"],
  version: 1,
  description: "Get, add and update data in Postgres",
  defaults: {
    name: "Postgres"
  },
  inputs: [import_n8n_workflow.NodeConnectionTypes.Main],
  outputs: [import_n8n_workflow.NodeConnectionTypes.Main],
  credentials: [
    {
      name: "postgres",
      required: true,
      testedBy: "postgresConnectionTest"
    }
  ],
  properties: [
    import_descriptions.oldVersionNotice,
    {
      displayName: "Operation",
      name: "operation",
      type: "options",
      noDataExpression: true,
      options: [
        {
          name: "Execute Query",
          value: "executeQuery",
          description: "Execute an SQL query",
          action: "Execute a SQL query"
        },
        {
          name: "Insert",
          value: "insert",
          description: "Insert rows in database",
          action: "Insert rows in database"
        },
        {
          name: "Update",
          value: "update",
          description: "Update rows in database",
          action: "Update rows in database"
        }
      ],
      default: "insert"
    },
    // ----------------------------------
    //         executeQuery
    // ----------------------------------
    {
      displayName: "Query",
      name: "query",
      type: "string",
      noDataExpression: true,
      typeOptions: {
        editor: "sqlEditor",
        sqlDialect: "PostgreSQL"
      },
      displayOptions: {
        show: {
          operation: ["executeQuery"]
        }
      },
      default: "",
      placeholder: "SELECT id, name FROM product WHERE quantity > $1 AND price <= $2",
      required: true,
      description: "The SQL query to execute. You can use n8n expressions or $1 and $2 in conjunction with query parameters."
    },
    // ----------------------------------
    //         insert
    // ----------------------------------
    {
      displayName: "Schema",
      name: "schema",
      type: "string",
      displayOptions: {
        show: {
          operation: ["insert"]
        }
      },
      default: "public",
      required: true,
      description: "Name of the schema the table belongs to"
    },
    {
      displayName: "Table",
      name: "table",
      type: "string",
      displayOptions: {
        show: {
          operation: ["insert"]
        }
      },
      default: "",
      required: true,
      description: "Name of the table in which to insert data to"
    },
    {
      displayName: "Columns",
      name: "columns",
      type: "string",
      displayOptions: {
        show: {
          operation: ["insert"]
        }
      },
      default: "",
      // eslint-disable-next-line n8n-nodes-base/node-param-placeholder-miscased-id
      placeholder: "id:int,name:text,description",
      // eslint-disable-next-line n8n-nodes-base/node-param-description-miscased-id
      description: "Comma-separated list of the properties which should used as columns for the new rows. You can use type casting with colons (:) like id:int."
    },
    // ----------------------------------
    //         update
    // ----------------------------------
    {
      displayName: "Schema",
      name: "schema",
      type: "string",
      displayOptions: {
        show: {
          operation: ["update"]
        }
      },
      default: "public",
      description: "Name of the schema the table belongs to"
    },
    {
      displayName: "Table",
      name: "table",
      type: "string",
      displayOptions: {
        show: {
          operation: ["update"]
        }
      },
      default: "",
      required: true,
      description: "Name of the table in which to update data in"
    },
    {
      displayName: "Update Key",
      name: "updateKey",
      type: "string",
      displayOptions: {
        show: {
          operation: ["update"]
        }
      },
      default: "id",
      required: true,
      // eslint-disable-next-line n8n-nodes-base/node-param-description-miscased-id
      description: 'Comma-separated list of the properties which decides which rows in the database should be updated. Normally that would be "id".'
    },
    {
      displayName: "Columns",
      name: "columns",
      type: "string",
      displayOptions: {
        show: {
          operation: ["update"]
        }
      },
      default: "",
      placeholder: "name:text,description",
      // eslint-disable-next-line n8n-nodes-base/node-param-description-miscased-id
      description: "Comma-separated list of the properties which should used as columns for rows to update. You can use type casting with colons (:) like id:int."
    },
    // ----------------------------------
    //         insert,update
    // ----------------------------------
    {
      displayName: "Return Fields",
      name: "returnFields",
      type: "string",
      requiresDataPath: "multiple",
      displayOptions: {
        show: {
          operation: ["insert", "update"]
        }
      },
      default: "*",
      description: "Comma-separated list of the fields that the operation will return"
    },
    // ----------------------------------
    //         Additional fields
    // ----------------------------------
    {
      displayName: "Additional Fields",
      name: "additionalFields",
      type: "collection",
      placeholder: "Add Field",
      default: {},
      options: [
        {
          displayName: "Mode",
          name: "mode",
          type: "options",
          options: [
            {
              name: "Independently",
              value: "independently",
              description: "Execute each query independently"
            },
            {
              name: "Multiple Queries",
              value: "multiple",
              description: "<b>Default</b>. Sends multiple queries at once to database."
            },
            {
              name: "Transaction",
              value: "transaction",
              description: "Executes all queries in a single transaction"
            }
          ],
          default: "multiple",
          description: 'The way queries should be sent to database. Can be used in conjunction with <b>Continue on Fail</b>. See <a href="https://docs.n8n.io/integrations/builtin/app-nodes/n8n-nodes-base.postgres/">the docs</a> for more examples'
        },
        {
          displayName: "Output Large-Format Numbers As",
          name: "largeNumbersOutput",
          type: "options",
          options: [
            {
              name: "Numbers",
              value: "numbers"
            },
            {
              name: "Text",
              value: "text",
              description: "Use this if you expect numbers longer than 16 digits (otherwise numbers may be incorrect)"
            }
          ],
          hint: "Applies to NUMERIC and BIGINT columns only",
          default: "text"
        },
        {
          displayName: "Query Parameters",
          name: "queryParams",
          type: "string",
          displayOptions: {
            show: {
              "/operation": ["executeQuery"]
            }
          },
          default: "",
          placeholder: "quantity,price",
          description: "Comma-separated list of properties which should be used as query parameters"
        }
      ]
    }
  ]
};
class PostgresV1 {
  constructor(baseDescription) {
    this.methods = {
      credentialTest: {
        async postgresConnectionTest(credential) {
          const credentials = credential.data;
          let connection;
          try {
            const { db } = await import_transport.configurePostgres.call(this, credentials, {});
            connection = await db.connect();
          } catch (error) {
            return {
              status: "Error",
              message: error.message
            };
          } finally {
            if (connection) {
              await connection.done();
            }
          }
          return {
            status: "OK",
            message: "Connection successful!"
          };
        }
      }
    };
    this.description = {
      ...baseDescription,
      ...versionDescription
    };
  }
  async execute() {
    const credentials = await this.getCredentials("postgres");
    const largeNumbersOutput = this.getNodeParameter(
      "additionalFields.largeNumbersOutput",
      0,
      ""
    );
    const { db, pgp } = await import_transport.configurePostgres.call(this, credentials, {
      largeNumbersOutput: largeNumbersOutput === "numbers" || largeNumbersOutput === "text" ? largeNumbersOutput : void 0
    });
    let returnItems = [];
    const items = this.getInputData();
    const operation = this.getNodeParameter("operation", 0);
    if (operation === "executeQuery") {
      const queryResult = await import_genericFunctions.pgQueryV2.call(this, pgp, db, items, this.continueOnFail());
      returnItems = queryResult;
    } else if (operation === "insert") {
      const insertData = await import_genericFunctions.pgInsertV2.call(this, pgp, db, items, this.continueOnFail());
      returnItems = insertData;
    } else if (operation === "update") {
      const updateItems = await (0, import_genericFunctions.pgUpdate)(
        this.getNodeParameter,
        pgp,
        db,
        items,
        this.continueOnFail()
      );
      returnItems = (0, import_genericFunctions.wrapData)(updateItems);
    } else {
      throw new import_n8n_workflow.NodeOperationError(
        this.getNode(),
        `The operation "${operation}" is not supported!`
      );
    }
    return [returnItems];
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PostgresV1
});
