// n8n/dist/services/orchestration.service.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrchestrationService = void 0;
const config_1 = require("@n8n/config");
const di_1 = require("@n8n/di");
const n8n_core_1 = require("n8n-core");
const config_2 = __importDefault(require("../config"));
const multi_main_setup_ee_1 = require("../scaling/multi-main-setup.ee");
let OrchestrationService = class OrchestrationService {
    constructor(instanceSettings, multiMainSetup, globalConfig) {
        this.instanceSettings = instanceSettings;
        this.multiMainSetup = multiMainSetup;
        this.globalConfig = globalConfig;
        this.isInitialized = false;
    }
    async init() {
        if (this.isInitialized)
            return;
        if (config_2.default.get('executions.mode') === 'queue') {
            const { Publisher } = await Promise.resolve().then(() => __importStar(require('../scaling/pubsub/publisher.service')));
            this.publisher = di_1.Container.get(Publisher);
            const { Subscriber } = await Promise.resolve().then(() => __importStar(require('../scaling/pubsub/subscriber.service')));
            this.subscriber = di_1.Container.get(Subscriber);
        }
        if (this.instanceSettings.isMultiMain) {
            await this.multiMainSetup.init();
        }
        else {
            this.instanceSettings.markAsLeader();
        }
        this.isInitialized = true;
    }
    async shutdown() {
        if (!this.isInitialized)
            return;
        if (this.instanceSettings.isMultiMain)
            await this.multiMainSetup.shutdown();
        this.publisher.shutdown();
        this.subscriber.shutdown();
        this.isInitialized = false;
    }
};
exports.OrchestrationService = OrchestrationService;
exports.OrchestrationService = OrchestrationService = __decorate([
    (0, di_1.Service)(),
    __metadata("design:paramtypes", [n8n_core_1.InstanceSettings,
        multi_main_setup_ee_1.MultiMainSetup,
        config_1.GlobalConfig])
], OrchestrationService);
